kong/conf_loader/init.lua

```diff
@@ -1536,31 +1547,59 @@ local function load(path, custom_conf, opts)
 
     loaded_vaults = setmetatable(vaults, _nop_tostring_mt)
 
-    local vault_conf = { loaded_vaults = loaded_vaults }
-    for k, v in pairs(conf) do
-      if sub(k, 1, 6) == "vault_" then
-        vault_conf[k] = infer_value(v, "string", opts)
-      end
-    end
+    if get_phase() == "init" then
+      local process_secrets = getenv("KONG_PROCESS_SECRETS")
```

Comment:
@Tieske,

It needs to be kept around in the master process as it is needed for `reload`.

Reply:
The ENV vars in general are not supported with `reload`, that is the master process env vars gets used.

Reply:
But what if a reload needs to update those values?

Reply:
@Tieske,

> But what if a reload needs to update those values?

You mean should reload re-fetch the latest secret values, that is interesting question. The problem is that it only executes init, so CLI cannot pass those values to master process, or it needs to somehow modify master process environment. We could setup a timer to re-fetch the secrets, but that is too later for e.g. postgres connection checks.

We will add though automatic rotation to postgres, so that should handle at least that part of it.

Reply:
thinking whether the file approach would be better...

Reply:
> are tests going to be added

This is quite hard as the `CE` doesn't have a vault that makes use of ngx.socket, but `EE` has one. So perhaps tests are added to `EE` or I need to create a test Vault.

Reply:
since reloading is hard and there's a history of bugs, I'd strongly advocate for adding tests. Maybe a test vault is the best approach.

Reply:
@Tieske, the tests were added.